streams/runners/storm/actions/src/main/java/com/hortonworks/streamline/streams/actions/storm/topology/StormTopologyActionsImpl.java

```diff
@@ -127,6 +136,35 @@ public void init (Map<String, String> conf) {
         }
         File f = new File (stormArtifactsLocation);
         f.mkdirs();
+
+        setupSecuredStormCluster(conf);
```

Comment:
Should not we be calling this method in deploy? I might be missing something here. But if we do it in init, how do we support the case that you mentioned where nimbus principal is different on different storm clusters. Also, one of them could be secured and other could be non-secured.